packages/itwin/grouping-mapping-widget/src/widget/components/groupsHelpers.ts

```diff
@@ -61,22 +67,25 @@ export const hideGroup = async (
 };
 
 const processGroupVisualization = async (
-  iModelConnection: IModelConnection,
-  group: Group,
+  overlappedElementGroupPairs: OverlappedElementGroupPairs,
   hiddenGroupsIds: Set<string>,
-  hilitedElementsQueryCache: React.MutableRefObject<Map<string, QueryCacheItem>>,
   doEmphasizeElements: boolean,
-  groupColor: string,
+  index: number,
   setNumberOfVisualizedGroups: (numberOfVisualizedGroups: number | ((numberOfVisualizedGroups: number) => number)) => void,
 ) => {
-  const result = await getHiliteIdsAndKeysetFromGroup(iModelConnection, group, hilitedElementsQueryCache);
-  const hilitedIds = result.ids;
-  overrideElements(hilitedIds, groupColor, FeatureOverrideType.ColorAndAlpha);
+  const hilitedIds = Array.from(overlappedElementGroupPairs.elementIds);
+  const redHsl = "hsl(0, 100%, 50%)";
```

Comment:
This looks like it could be moved to a function. Above, in the function `getGroupColor` the same string is parameterized.

Reply:
Moved around some logic.

Reply:
Even after moving the logic around these is no function that generates the hsl string.

Reply:
Done